src/renderers/webgl/WebGLShadowMap.js

```diff
@@ -70,10 +71,12 @@ function WebGLShadowMap( _renderer, _lights, _objects, capabilities ) {
 
 		var useMorphing = ( i & _MorphingFlag ) !== 0;
 		var useSkinning = ( i & _SkinningFlag ) !== 0;
+		var useInstancing = ( i * _InstancingFlag ) !==0;
```

Comment:
ok ill look a bit through your comments, and play around with updates, without an instance of Object3d, there should be some dirt flags in there too? Or should this just be left static? If so then this is ready to go, there are no changes to any of the files except for the three shader snippets. What i'd do maybe is extend the define mechanism a little bit. Right now the piggybacking on `uv_pars_vertex` seems to work, but it feels like there should be a dedicated `global_pars_vertex` since common is used in fragment shaders as well...

Could that be a part of this commit or a separate thing?